runtime/compilers/rillv1/parser.go

```diff
@@ -563,6 +577,24 @@ func (p *Parser) parsePaths(ctx context.Context, paths []string) error {
 		if pathIsRillYAML(b) {
 			return 1
 		}
+
+		// Sort .env files by path depth with always root first. This ensures that .env files are parsed in order from root to deepest nested
+		aIsDotEnv := pathIsDotEnv(a)
+		bIsDotEnv := pathIsDotEnv(b)
+		if aIsDotEnv && !bIsDotEnv {
+			return -1
+		}
+		if !aIsDotEnv && bIsDotEnv {
+			return 1
+		}
+		if aIsDotEnv && bIsDotEnv {
+			aDepth := strings.Count(a, "/")
+			bDepth := strings.Count(b, "/")
+			if aDepth != bDepth {
+				return aDepth - bDepth
+			}
+		}
```

Comment:
I don't think this matters when you store the values as separate sub-maps of `p.DotEnv`? Instead the precedence would need to be applied in the implementation of `GetDotEnv`.